airbyte-cdk/python/airbyte_cdk/sources/file_based/config/file_based_stream_config.py

```diff
@@ -82,6 +97,9 @@ def validate_encoding(cls, v: str) -> str:
         return v
 
 
+VALID_FILE_TYPES = {"csv": CsvFormat, "parquet": ParquetFormat, "jsonl": JsonlFormat}
```

Comment:
something to think about as we implement https://github.com/airbytehq/airbyte/issues/28254: should the parser know about the spec? Basically they could be "parser components" analogous to the declarative components